packages/expo/src/hooks/useAppleSignIn.ts

```diff
@@ -0,0 +1,159 @@
+import * as AppleAuthentication from 'expo-apple-authentication';
+import { Platform } from 'react-native';
+
+import { errorThrower } from '../utils/errors';
+
+export type StartAppleSignInFlowParams = {
+  unsafeMetadata?: SignUpUnsafeMetadata;
+};
+
+export type StartAppleSignInFlowReturnType = {
+  createdSessionId: string | null;
+  setActive?: SetActive;
+  signIn?: SignInResource;
+  signUp?: SignUpResource;
+};
+
+/**
+ * Hook for native Apple Sign-In on iOS using expo-apple-authentication.
+ *
+ * This hook provides a simplified way to authenticate users with their Apple ID
+ * using the native iOS Sign in with Apple UI. The authentication flow automatically
+ * handles the ID token exchange with Clerk's backend and manages the transfer flow
+ * between sign-in and sign-up.
+ *
+ * @example
+ * ```tsx
+ * import { useAppleSignIn } from '@clerk/clerk-expo';
+ * import { Button } from 'react-native';
+ *
+ * function AppleSignInButton() {
+ *   const { startAppleSignInFlow } = useAppleSignIn();
+ *
+ *   const onPress = async () => {
+ *     try {
+ *       const { createdSessionId, setActive } = await startAppleSignInFlow();
+ *
+ *       if (createdSessionId && setActive) {
+ *         await setActive({ session: createdSessionId });
+ *       }
+ *     } catch (err) {
+ *       console.error('Apple Sign-In error:', err);
+ *     }
+ *   };
+ *
+ *   return <Button title="Sign in with Apple" onPress={onPress} />;
+ * }
+ * ```
+ *
+ * @requires expo-apple-authentication - Must be installed as a peer dependency
+ * @platform iOS - This hook only works on iOS. On other platforms, it will throw an error.
+ *
+ * @returns An object containing the `startAppleSignInFlow` function
+ */
+export function useAppleSignIn() {
+  const { signIn, setActive, isLoaded: isSignInLoaded } = useSignIn();
+  const { signUp, isLoaded: isSignUpLoaded } = useSignUp();
+
+  async function startAppleSignInFlow(
+    startAppleSignInFlowParams?: StartAppleSignInFlowParams,
+  ): Promise<StartAppleSignInFlowReturnType> {
+    // Check platform compatibility
+    if (Platform.OS !== 'ios') {
+      return errorThrower.throw(
+        'Apple Sign-In is only available on iOS. Please use the web-based OAuth flow (useSSO with strategy: "oauth_apple") on other platforms.',
+      );
+    }
+
+    if (!isSignInLoaded || !isSignUpLoaded) {
+      return {
+        createdSessionId: null,
+        signIn,
+        signUp,
+        setActive,
+      };
+    }
+
+    // Check if Apple Authentication is available on the device
+    const isAvailable = await AppleAuthentication.isAvailableAsync();
+    if (!isAvailable) {
+      return errorThrower.throw('Apple Authentication is not available on this device.');
+    }
+
+    try {
+      // Generate a cryptographically secure nonce for the Apple Sign-In request (required by Clerk)
+      // Lazy load expo-crypto to avoid import issues on web
+      const Crypto = await import('expo-crypto');
```

Comment:
stuff like this is why I think it might be better to write this as a platform-specific file